bhop.s

```diff
@@ -2353,33 +2359,97 @@ dpcm_cut:
         sta $4011
 dpcm_release:
         lda dpcm_status
-        .if ::BHOP_ZSAW_ENABLED
-        and #($FF - (DPCM_ZSAW_ENABLED))
-        .endif
         and #($FF - (DPCM_ENABLED))
         sta dpcm_status
 
 check_for_inactive:
-        .if ::BHOP_ZSAW_ENABLED
-        ; Only take action if Z-Saw is disabled...
-        lda dpcm_status
-        and #DPCM_ZSAW_ENABLED
-        bne done
+        ; Only take action if virtual Z channels are disabled...
+        .if ::BHOP_ZSAW_ENABLED .or ::BHOP_ZPCM_ENABLED
 
         ; See if that DPCM playback has finished:
         lda $4015
         and #%00010000
         bne done
 
-        ; If it has, enable the Z-Saw channel
+        ; If it has, enable virtual Z channels
         ; to initiate playback on the next tick
+            .if ::BHOP_ZSAW_ENABLED
+            jsr zsaw_enable
+            .elseif ::BHOP_ZPCM_ENABLED
+            jsr zpcm_enable
+            .endif
+        .endif
+
+        rts
+.endproc
+
+.if ::BHOP_ZPCM_ENABLED
+; request to disable ZPCM
+.proc zpcm_disable
+        ; check if ZPCM is already disabled first
         lda dpcm_status
-        ora #DPCM_ZSAW_ENABLED
+        and #DPCM_ZPCM_ENABLED
+        beq done
+        jsr BHOP_ZPCM_DISABLE_ROUTINE
```

Comment:
This jsr needs to be guarded on BHOP_CONFLICT_AVOIDANCE, otherwise the symbol is never defined and the project will fail to compile. Same with the enable routine below.